microscopium/features.py

```diff
@@ -216,6 +218,82 @@ def object_features(bin_im, im, erode=2, sample_size=None, random_seed=None):
     return fs, names
 
 
+def haralick_features(im, prop_names=None, distances=[5], angles=[0],
+                      levels=256, symmetric=False, normed=False):
```

Comment:
Where do the defaults for `distances` and `angles` come from? I'd suggest something a bit more thorough by default, and on a log scale, say `distances=[2, 4, 8]`, and `angles=np.arange(8) * np.pi / 4`. What do you think?

Reply:
They are entirely arbitrary (I took them from [the example](https://scikit-image.org/docs/stable/auto_examples/features_detection/plot_glcm.html#sphx-glr-auto-examples-features-detection-plot-glcm-py), but that's not necessarily appropriate here). 

I couldn't work out from the `greycomatrix` docstring what a sensible value should look like! I'm in favour of a PR to scikit-image adding more detail to the docstring once I understand it.

> I'd suggest something a bit more thorough by default, and on a log scale, say `distances=[2, 4, 8]`, and `angles=np.arange(8) * np.pi / 4`. What do you think?

Are you suggesting those exact numbers?

The angles sound good, a log scale sounds good, not sure how to judge the exact pixel values for distances.

Reply:
well, if you do log-scale, then the only question is start and stop. I thought 1, 2, 4, 8 might be a lot of extra work (33% more) for not a lot of benefit (nearby pixels are usually extremely correlated anyway). So I think [2, 4, 8] is probably a good default. 

grey co-occurrence matrices are kind of what they sound: you're looking at correlation of pixel intensity values at 2, 4, 8 pixels away, at the given angles. Different textures will have different correlation values.

Come to think of it, since we are probably looking at rotation-invariant things, we can probably leave angles=[0] as the default! :joy: